src/OrchardCore.Modules/OrchardCore.Liquid/Services/SlugService.cs

```diff
@@ -1,75 +1,70 @@
+using System;
 using System.Globalization;
+using System.Linq;
 using System.Text;
 
 namespace OrchardCore.Liquid.Services
 {
     public class SlugService : ISlugService
     {
+        private const char Hyphen = '-';
+
+        private static readonly char[] _allowedSymbols = new char[] { Hyphen, '_', '~' };
+
         public string Slugify(string text)
         {
-            if (string.IsNullOrEmpty(text))
+            if (String.IsNullOrEmpty(text))
             {
                 return text;
             }
 
-            var sb = new StringBuilder();
+            var slug = new StringBuilder();
```

Comment:
if you are optimizing it, wouldn't you use `ZString` ?

Reply:
I will try this

Reply:
I run this using ZString the results are:

|          Method |     Mean |     Error |    StdDev |   Gen 0 | Allocated |
|---------------- |---------:|----------:|----------:|--------:|----------:|
| EvaluateSlugify | 4.659 us | 0.1532 us | 0.0084 us | 31.2500 |     64 KB |

Also I run it again without

|          Method |     Mean |     Error |    StdDev |  Gen 0 | Allocated |
|---------------- |---------:|----------:|----------:|-------:|----------:|
| EvaluateSlugify | 1.423 us | 0.4246 us | 0.0233 us | 0.2174 |     456 B |

Seems ZString allocate less memory but I don't have much knowlege about the other factors, so @sebastienros could you please let me know which is better and what other factors mean

Reply:
Pooling will have its cost, allocations reduced but more machinery running.  The sample you showed is quite short string, is it the normal case. Also if this code is run say 5-10 times a page request the benefits would so with more stress - now CPU is not needed for GC as sample runs so fast and does little - load benchmark would probably show more difference.

Reply:
Thanks for your information Marko, I may create another benchmark for a long string and see how it iss going

Reply:
OK now I'm reading the table again (I though they both were B) the ZString allocation looks really odd, can you show the diff for code you use with ZString and without?

Reply:
I just use 

```csharp
var slug = ZString.CreateStringBuilder();
```

instead of
```csharp
var slug = new StringBuilder();
```

Reply:
StringBuilder

|                      Method |     Mean |    Error |   StdDev |  Gen 0 | Allocated |
|---------------------------- |---------:|---------:|---------:|-------:|----------:|
| EvaluateSlugifyWithLongSlug | 16.97 us | 1.529 us | 0.084 us | 2.9907 |      6 KB |

ZString.CreateStringBuilder

|                      Method |     Mean |    Error |   StdDev |   Gen 0 | Allocated |
|---------------------------- |---------:|---------:|---------:|--------:|----------:|
| EvaluateSlugifyWithLongSlug | 20.61 us | 5.802 us | 0.318 us | 32.2571 |     68 KB |

Seems the current implementation is fine, unless there's something I don't know about ZString

Reply:
Yes, you were missing `using` for the ZString creation. I've commented my results into PR main discussion.